Services/UI/OJS.Services.Ui.Business/Implementations/ProblemsBusinessService.cs

```diff
@@ -137,8 +137,12 @@ public async Task<ProblemSearchServiceResultModel> GetSearchProblemsByName(Searc
         {
             var modelResult = new ProblemSearchServiceResultModel();
 
-            var allProblemsQueryable = this.problemsData.GetAllNonDeletedProblems()
-                .Where(p => p.Name.Contains(model.SearchTerm!));
+            var allProblemsQueryable = this.problemsData
+                .GetAllNonDeletedProblems()
+                .Where(p => p.Name.Contains(model.SearchTerm!) &&
```

Comment:
Avoid using `!` where possible (`model.SearchTerm!`)

Reply:
This part of the query was there before my changes were introduced. Should the ! be removed? Will take note for this usage the future.

Reply:
Should be reworked to not throw null reference exception.

Reply:
Fixed by using ?? operator instead of null-suppression operator. 
Additional note: Technically the searchTerm can never be empty since the front end is making validation and displaying error message for the longitude of the search term. 

![image](https://github.com/SoftUni-Internal/OpenJudgeSystem/assets/100528439/ec43b947-d83b-4fe2-a22c-4758dd472fb5)

Reply:
Great. I suppose it cannot be, but nevertheless on the back end we should not depend on front end validations.